perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -1095,6 +1096,43 @@ private void removeDirectMember(PerunSession sess, Group group, Member member) t
 		if (!VosManager.MEMBERS_GROUP.equals(group.getName())) {
 			recalculateMemberGroupStatusRecursively(sess, member, group);
 		}
+		addMemberToGroupsFromTriggerAttribute(sess, group, member);
```

Comment:
You should call this only if member was removed totally (line 1086/87 in this file)

Reply:
I have added a condition to the end of the method so it should work as you wanted.